syntax = "proto3";

import "bob_emploi/frontend/api/options.proto";

package bayes.bob;

message Event {
  // Title of this event.
  string title = 1;

  // Link to the event's page.
  string link = 2;

  // A set of filter IDs: do not show the event if one these
  // filters returns false for the project.
  repeated string filters = 3 [(string_format) = SCORING_MODEL_ID];

  // Name of the organiser of the event.
  string organiser = 4;

  // Start date of the event in international format, e.g. 2017-08-11.
  string start_date = 5;

  // Short description of the event. This may use markdown.
  string description = 6;

  // Short text describing the date and time of the event, e.g. "mercredi 7
  // février de 14h à 16h".
  string timing_text = 7;

  // Name of the city in which the event will take place.
  string city_name = 8;

  // Latitude of the place in which the event will take place.
  float latitude = 9;

  // Longitude of the place in which the event will take place.
  float longitude = 10;
}

message Events {
  repeated Event events = 1;
}

message CloseByEvents {
  // A set of events not far from the requested location.
  repeated Event events = 1;

  // Name of a city close to which the events are taking place. Note that if
  // this is empty, events can actually be anywhere.
  string city = 2;
}
